src/bptl/dashboard/filters.py

```diff
@@ -19,10 +19,21 @@ class TaskFilter(FilterSet):
         label="Type",
         widget=forms.CheckboxSelectMultiple,
     )
+    instance_id = django_filters.CharFilter(
+        method="filter_by_process_instance_id",
+        label="Process Instance (ID)",
+        widget=forms.TextInput,
+    )
 
     class Meta:
         model = BaseTask
-        fields = ("status", "topic_name", "engine_type")
+        fields = ("status", "topic_name", "engine_type", "instance_id")
+
+    def filter_by_process_instance_id(self, queryset, name, value: str) -> QuerySet:
+        if not value:
+            return queryset
+        qs = queryset.filter(externaltask__instance_id__iexact=value)
```

Comment:
maybe allow searching on a part of an ID even?

I also think that you can declare this lookup directly in the filter rather than using `method` argument

```suggestion
        qs = queryset.filter(externaltask__instance_id__icontains=value)
```